### PR TITLE
Migrate from kotlin-android-extensions to kotlin-parcelize

### PR DESCRIPTION
**Pull request by tonidero**
### Description
https://revenuecats.atlassian.net/browse/CSDK-123

`kotlin-android-extensions` was deprecated a while back. We only used the `parcelize` functionalities of that plugin, which were moved to the `kotlin-parcelize` plugin. With this PR we use that new plugin. In order to use the new plugin I had to update our kotlin version from 1.4.10 to 1.4.20. Didn't see any changes there that should cause problems 

The changes mostly involved changing the imports to the new package.

### CODE REVIEW

**Review by aboedo — COMMENTED** [review]
Changes look good to me, but waiting for @beylmk or @vegaro to give the thumbs up

**Review by aboedo — APPROVED** [review]
Oh, just noticed that these changes were also a part of https://github.com/RevenueCat/purchases-android/pull/474. Looks safe to merge

**Comment by tonidero** [comment]
Ahh I didn't notice that PR already had these changes. That PR bumps a bunch of extra versions which would be good to get as well, but there shouldn't be many conflicts after merging this, except maybe on the kotlin version. So will go ahead and merge it.

**Comment by vegaro** [comment]
Thanks for doing this. I kept that #474 PR as draft because the number of changes were growing so much and got blocked by a weird error and decided to come back to it in the future.

File: public/src/main/java/com/revenuecat/purchases/Offering.kt
Comment (vegaro) [inline public/src/main/java/com/revenuecat/purchases/Offering.kt:9]: I believe this one also needs to be changed like I did on https://github.com/RevenueCat/purchases-android/pull/474/files#diff-869b813b31b540b6a7fbf9b7fa2d5bffe1011dcda27ce036be533907e9a5b47fR9, but I am not sure

Comment (tonidero) [reply public/src/main/java/com/revenuecat/purchases/Offering.kt:9]: Good catch! I would have thought this wouldn't compile... Will fix it on a followup PR. Thanks!

**Pull request merged**
